src/buildstream_plugins/elements/autotools.yaml

```diff
@@ -93,20 +94,39 @@ config:
   - |
     %{autogen}
   - |
-    %{configure}
+    if [ -n "%{build-dir}" ]; then
+      mkdir %{build-dir}
+      pushd %{build-dir}
+      %{configure}
+      popd
+    else
+      %{configure}
+    fi
 
   # Commands for building the software
   #
   build-commands:
   - |
-    %{make}
+    if [ -n "%{build-dir}" ]; then
+      pushd %{build-dir}
+      %{make}
+      popd
```

Comment:
Either way, it's better to use the `-C` switch with make instead of this dance

Reply:
Yeah, I guss it's clearer

Reply:
FWIW this is what we're doing right now in freedesktop-sdk, this is mostly copy-paste from that. The difference is what freedesktop-sdk does relies on the fact that bst1 resets working directory between script configure-commands, build-commands and install-commands.

Reply:
I'm not sure if it works correctly though. The point is that element author is supposed to be able to do `make -C subcomponent` and this works *inside build-dir*. So we need to change working directory outside make.

Reply:
> As far as I've seen bst2 does not.

This is an important detail we have thus far omitted from the [porting guide](https://docs.buildstream.build/master/porting_project.html).

Opening https://github.com/apache/buildstream/issues/1708 to track this.

Reply:
It seems the more portable way to do this is using subshells

Reply:
Right, subshell strategy definitely works. What is iffy here is whether it would be cleaner to pre-create commands-subdir if it doesn't exist, run autogen in subshell in conf-root (where it normally has to be run) and tell consumer that they can set build-dir through `command-subdir: build-dir`.